dts: bindings: dai: Change the property names in the DTS

rruuaanng opened the pull request:
Rename the following properties in bindings and DTS:

-- dai_id =>dai-id
-- afe_name => afe-name
-- agent_disable => agent-disable
-- ch_num => ch-num
-- mono_invert => mono-invert
-- quad_ch => quad-ch
-- int_odd => int-odd

fabiobaltieri commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -201,6 +201,17 @@ DAC
+DAI
+===
+
+* Renamed the devicetree property ``dai_id`` to ``dai-id``.
+* Renamed the devicetree property ``afe_name`` to ``afe-name``.
+* Renamed the devicetree property ``agent_disable`` to ``agent-disable``.
+* Renamed the devicetree property ``ch_num`` to ``ch-num``.
+* Renamed the devicetree property ``mono_invert`` to ``mono-invert``.
+* Renamed the devicetree property ``quad_ch`` to ``quad-ch``.
+* Renamed the devicetree property ``int_odd`` to ``int-odd``.
```
same as the other one, this will have to wait for 4.2

rruuaanng replied:
+1

dbaluta commented on boards/mediatek/mt8186/afe-mt8186.dts:
```diff
@@ -1,7 +1,7 @@
 	afe_dl1: afe_dl1 {
```
Please explain WHY this change is needed. Why do we need to use dai_id instead of dai-id.

rruuaanng replied:
See it :) https://github.com/zephyrproject-rtos/zephyr/issues/83741

dbaluta replied:
I know this, but please mention it in the commit message. A commit message should tell more about WHY the change is needed and less WHAT does the change do. 

The WHAT should be obvious from the code :).

rruuaanng replied:
I included a note about the RFC in my commit message.

rruuaanng committed:
dts: bindings: dai: Change the property names in the DTS

Rename the following properties in bindings and DTS
to implement RFC-53506:

-- dai_id =>dai-id
-- afe_name => afe-name
-- agent_disable => agent-disable
-- ch_num => ch-num
-- mono_invert => mono-invert
-- quad_ch => quad-ch
-- int_odd => int-odd

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
doc: Add a new migration guide entry to DAI

Add the following property renaming entries to
the migration:

* dai_id
* afe_name
* agent_disable
* ch_num
* mono_invert
* quad_ch
* int_odd

Signed-off-by: James Roy <rruuaanng@outlook.com>

kartben commented:
@rruuaanng needs to be rebased

maass-hamburg commented:
@kartben it seems like this PR with 394c78538fc8bba5e0535ad36c7d9183de968e26 removed some unrelated entries from the migration guide?

rruuaanng commented:
@kartben Of course. Oops, looks like I removed something I shouldn't have when using the conflict tool.
